src/compiler/commandLineParser.ts

```diff
@@ -872,6 +872,21 @@ namespace ts {
             description: Diagnostics.Include_modules_imported_with_json_extension
         },
 
+        {
+            name: "metadataDecorator",
+            type: "string",
+            category: Diagnostics.Advanced_Options,

```

Comment:
Shouldnt this have affectsEmit set to true?

Reply:
I wasn't sure. `jsxFactory` and `jsxFragmentFactory` (which is what I modeled this after), do not have it set but perhaps they should as well.

Reply:
Yeah, I think that sounds accurate.

Reply:
I've updated `metadataDecorator`. I'm holding off on `jsxFactory` and `jsxFragmentFactory` for a separate PR if they need to be added.